src/invoices/InvoicesList/InvoiceLine/InvoiceLineDetailsLayer.js

```diff
@@ -51,19 +58,32 @@ class InvoiceLineDetailsLayer extends Component {
 
   render() {
     const {
+      mutator,
       resources,
     } = this.props;
+    const { isTagsPaneOpened } = this.state;
+
     const invoiceLine = this.getInvoiceLine();
     const hasLoaded = get(resources, 'invoiceLine.hasLoaded');
 
     return hasLoaded
       ? (
-        <InvoiceLineDetails
-          closeInvoiceLine={this.closeInvoiceLine}
-          deleteInvoiceLine={this.deleteInvoiceLine}
-          goToEditInvoiceLine={this.goToEditInvoiceLine}
-          invoiceLine={invoiceLine}
-        />
+        <Fragment>
+          <InvoiceLineDetails
+            closeInvoiceLine={this.closeInvoiceLine}
+            deleteInvoiceLine={this.deleteInvoiceLine}
+            goToEditInvoiceLine={this.goToEditInvoiceLine}
+            invoiceLine={invoiceLine}
+            tagsToggle={this.toggleTagsPane}
+          />
+          {isTagsPaneOpened && (
+            <TagsContainer
+              putMutator={mutator.invoiceLine.PUT}
```

Comment:
so you've passed the whole `mutator` for only one reason - to pass down here its `invoiceLine` property? Usually we pass down callback function

Reply:
In this very case, when you really want to see mutator put function below, - it's fine to pass mutator function, but no any reason to pass whe whole `mutator` object here.

Reply:
mutator passes from Layer to View component